src/presetsController.ts

```diff
@@ -0,0 +1,844 @@
+                    filteredKit.preferredGenerator?.toolset === kit.preferredGenerator?.toolset) {
+                  // Found same generator in the filtered list
+                  found = true;
+                  break;
+                }
+              }
+            }
+            if (!found) {
+              filteredKits.push(kit);
+            }
+          }
+
+          log.debug(localize('start.selection.of.compilers', 'Start selection of compilers. Found {0} compilers.', filteredKits.length));
+
+          interface KitItem extends vscode.QuickPickItem {
+            kit: Kit;
+          }
+          log.debug(localize('opening.compiler.selection', 'Opening compiler selection QuickPick'));
+          // Generate the quickpick items from our known kits
+          const getKitName = (kit: Kit) => {
+            if (kit.name === SpecialKits.ScanForKits) {
+              return `[${localize('scan.for.compilers.button', 'Scan for compilers')}]`;
+            } else if (kit.visualStudio && !kit.compilers) {
+              const hostTargetArch = kitHostTargetArch(kit.visualStudioArchitecture!, kit.preferredGenerator?.platform);
+              return `${(kit.preferredGenerator?.name || 'Visual Studio')} ${hostTargetArch}`;
+            } else {
+              return kit.name;
+            }
+          };
+          const item_promises = filteredKits.map(
+              async (kit): Promise<KitItem> => ({
+                label: getKitName(kit),
+                description: await descriptionForKit(kit, true),
+                kit,
+              }),
+          );
+          const quickPickItems = await Promise.all(item_promises);
+          const chosen_kit = await vscode.window.showQuickPick(quickPickItems,
+                                                               {placeHolder: localize('select.a.compiler.placeholder', 'Select a Kit for {0}', this.folder.name)});
+          if (chosen_kit === undefined) {
+            log.debug(localize('user.cancelled.compiler.selection', 'User cancelled compiler selection'));
+            // No selection was made
+            return false;
+          } else {
+            if (chosen_kit.kit.name == SpecialKits.ScanForKits) {
+              await KitsController.scanForKits(this._cmakeTools);
+              preset.setCompilers(this._kitsController.availableKits);
+              return false;
+            } else {
+              log.debug(localize('user.selected.compiler', 'User selected compiler {0}', JSON.stringify(chosen_kit)));
+              newPreset = {
+                name: '__placeholder__',
+                displayName: chosen_kit.kit.name,
+                description: chosen_kit.description,
+                generator: chosen_kit.kit.preferredGenerator?.name,
+                toolset: chosen_kit.kit.preferredGenerator?.toolset,
+                architecture: chosen_kit.kit.preferredGenerator?.platform,
+                binaryDir: '${sourceDir}/out/build/${presetName}',
```

Comment:
This is at odds with the default value of `cmake.buildDirectory`.  We don't normally add `out` to the path.  I'll send Erika an email and ask what she thinks.

Reply:
We can leave it as 'out'